pkg/github/status_service.go

```diff
@@ -28,32 +29,33 @@ func NewStatusService(client *github.Client, log log.Logger, change scm.Reposito
 }
 
 // Success marks given change as a success.
-func (s *StatusService) Success(reason string) error {
-	return s.setStatus(StatusSuccess, reason)
+func (s *StatusService) Success(reason, targetURL string) error {
```

Comment:
I was wondering - since we already have a concept of `StatusContext` which is part of this service, maybe we could have target URLs as part of it? Then we can keep simple signatures with only the `reason` to be passed?

Reply:
I'm not sure if the `StatusContext` is the best place for it. If we moved it there, then we should move the `reason` there as well. When you take a look at the usage, then you can see that the `TargetURL` is different for each request - every details link that is related only to one reason

Reply:
My main rationale for that is that those links are right now static so we have them at the moment of service instantiation. We have 3 different links we can use within private functions (that's why I said target URL**s** in the original comment).

Reply:
> My main rationale for that is that those links are right now static so we have them at the moment of service instantiation.

you can say the same about the reasons

> We have 3 different links we can use within private functions (that's why I said target URLs in the original comment).

3 different links for the success and one link for failure. I don't understand how would you figure out which "success" link should be used on the level of the `StatusService` or `StatusContext`. You need to provide it along with the reason. 

btw, the `StatusContext` contains `BotName` and `PluginName` - both of them are same for all "set-status" requests. Details link, as well as the reason, is different for every state.

Reply:
> 3 different links for the success and one link for failure

Right missed that part. Could be early hour ;) Let's stick to what we have now. Maybe giving `targetURL` a different name like `detailsLink`? Also could make it optional, so we can call it with or without it instead of passing `""` in case of latter.

Reply:
renamed here 7043f42 

> Also could make it optional, so we can call it with or without it instead of passing "" in case of latter.

I would keep it mandatory for both statuses failure and success. I don't think that there will be a use case when you need to set the state without providing the details link.